engine/src/object.cpp

```diff
@@ -5571,6 +5571,46 @@ static void MCDeletedObjectPoolDestroy(MCDeletedObjectPool *p_pool)
     MCMemoryDelete(p_pool);
 }
 
+static MCDeletedObjectPool* MCDeletedObjectPoolCleanup(MCDeletedObjectPool *p_pool)
+{
+	while (p_pool != nil && p_pool->defunct && p_pool->references == 0)
+	{
+		MCDeletedObjectPool *t_this_pool = p_pool;
+		p_pool = p_pool->parent;
+		MCDeletedObjectPoolDestroy(t_this_pool);
+	}
+	
+	return p_pool;
+}
+
+static MCDeletedObjectPool* MCDeletedObjectPoolRelease(MCDeletedObjectPool *p_pool)
+{
+	if (p_pool == nil)
+		return nil;
+	
+	p_pool->references -= 1;
+	
+	return MCDeletedObjectPoolCleanup(p_pool);
+}
+
+static MCDeletedObjectPool* MCDeletedObjectPoolMakeDefunct(MCDeletedObjectPool* p_pool)
+{
+	if (p_pool == nil)
+		return nil;
+	
+	p_pool->defunct = true;
+	
+	return MCDeletedObjectPoolCleanup(p_pool);
+}
+
+static MCDeletedObjectPool* MCDeletedObjectPoolGetNonDefunct(MCDeletedObjectPool *p_pool)
```

Comment:
Change this to Unwind as suggested below, and merge with the body of Release (rather than calling Release a sub-routine - the main work is done in Cleanup which is fine).